src/test/java/com/github/packageurl/JsonTestData.java

```diff
@@ -0,0 +1,109 @@
+package com.github.packageurl;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Collections;
+import java.util.Map;
+import org.jspecify.annotations.Nullable;
+
+class JsonTestData {
+    private String description = "";
+    private String purl = "";
+    private String canonicalPurl = "";
```

Comment:
Why initialize these here? It looks like `description` and `purl` are always set and can't be null already. The canonical purl can be null if it's not valid.

Reply:
Sure, it is not useful, but it removes IDE warnings about nullability (and it is not in production code).

I fixed `canonicalPurl` in https://github.com/package-url/packageurl-java/pull/202/commits/143c7f8978c5c196bbb14f56e5ed099965738a86